qcodes/instrument_drivers/Spectrum/M4i.py

```diff
@@ -777,22 +786,41 @@ def start_acquisition(self, mV_range, memsize, posttrigger_size=None, verbose=0)
 
         return {'memsize': memsize, 'numch': numch, 'mV_range': mV_range}
 
-    def _transfer_buffer_numpy(self, memsize, numch):
-        """ Transfer buffer to numpy array """
+    def _transfer_buffer_numpy(self, memsize, numch, bytes_per_sample=2):

```

Comment:
do i understand correctly that this method not only transfers the data from the buffer to the PC, but also triggers the acquisition of the data (because of the `self.general_command(...)` call)? If so, then the name of this method is confusing. I'd suggest to call it `_acquire_and_transfer` or similar. The fact that a numpy array is returned can be documented in the docstrings, and (even better) in the type annotation of the method.

Reply:
No, the acquisition starts with other commands (e.g. M2CMD_CARD_START). It is true the methods blocks untill the acquisition is complete. I added a type annotation for this method (but will not annotate the entire driver)